memdx/ops_crud_test.go

```diff
@@ -2865,6 +2865,262 @@ func TestOpsCrudValueTooLarge(t *testing.T) {
 	}
 }
 
+func TestOpsCrudOnBehalfOf(t *testing.T) {
+	testutils.SkipIfShortTest(t)
+
+	cli := createTestClient(t)
+
+	type test struct {
+		Op   func(key []byte, opsCrud OpsCrud, cb func(interface{}, error)) (PendingOp, error)
+		Name string
+	}
+
+	/*
+		Since it isn't possible to inspect what particular user an operation was executed
+		under, we test on-behalf-of execution by setting the user to something invalid and
+		then validating that the operation is rejected with access denied.
+	*/
+
+	tests := []test{
+		{
+			Name: "Get",
+			Op: func(key []byte, opsCrud OpsCrud, cb func(interface{}, error)) (PendingOp, error) {
+				return opsCrud.Get(cli, &GetRequest{
+					Key:       key,
+					VbucketID: defaultTestVbucketID,
+					CrudRequestMeta: CrudRequestMeta{
+						OnBehalfOf: "invalid-user",
```

Comment:
What is the purpose of the `OnBehalfOfField`?

Reply:
OnBehalfOf allows you to authenticate as one user, but then execute an operation in the context of another.  We use it in CNG so that CNG can authenticate as one user, but then execute the operation under the context of some other authenticated user (the one from the request).